File: 0370-range-addition/0370-range-addition.js
/**
 * @param {number} length
 * @param {number[][]} updates
 * @return {number[]}
 */
var getModifiedArray = function(length, updates) {
    const diff = new Array(length + 1).fill(0);
    for(const x of updates) {
        diff[x[0]] += x[2]
        diff[x[1] + 1] -= x[2]
    }
    
    let val = 0;
    const res = new Array(length);
    
    for(let i = 0; i < length; ++i) {
        val += diff[i];
        res[i] = val;
    }
    
    return res;
};

File: 0370-range-addition/0370-range-addition.cpp
class Solution {
public:
    vector<int> getModifiedArray(int length, vector<vector<int>>& updates) {
        vector<int> diff(length + 1);
        for(auto u : updates) {
            diff[u[0]] += u[2];
            diff[u[1] + 1] -= u[2];
        }
        
        vector<int> res(length);
        int val = 0;
        for(int i = 0; i < length; ++i) {
            val += diff[i];
            res[i] = val;
        }
        return res;
    }
};